base/ca/src/main/java/com/netscape/cmscore/dbs/CertificateRepository.java

```diff
@@ -1629,57 +1602,57 @@ public X509CertImpl[] getX509Certificates(String filter)
      */
     public Enumeration<CertRecord> getValidCertificates(String from, String to)
             throws EBaseException {
-        Vector<CertRecord> v = new Vector<>();
+        ArrayList<CertRecord> a = new ArrayList<>();
 
         // 'from' determines 'jumpto' value
         // 'to' determines where to stop looking
 
-        String ldapfilter = "(certstatus=VALID)";
+        StringBuilder ldapfilter = new StringBuilder();
 
-        String fromVal = "0";
-        try {
-            if (from != null) {
-                new BigInteger(from);
-                fromVal = from;
+        if (from != null && !from.isBlank()) {
+            ldapfilter.append("(&(")
+                .append(CertRecord.ATTR_ID)
+                .append(">=")
+                .append(from)
+                .append(")");
+        }
+        if (to != null && !to.isBlank()) {
+            if(ldapfilter.isEmpty()) {
+                ldapfilter.append("(&");
             }
-        } catch (Exception e1) {
-            // from is not integer
+            ldapfilter.append("(")
+                .append(CertRecord.ATTR_ID)
+                .append("<")
```

Comment:
Officially LDAP does not support a `<` operator, so `(a<b)` needs to be written as `(!(b>=a))`, but I don't know whether DS actually supports `<`.